config/discover.ml

```diff
@@ -0,0 +1,380 @@
+  "CURLOPT_MAXFILESIZE_LARGE";
+  "CURLOPT_MAXREDIRS";
+  "CURLOPT_MAX_RECV_SPEED_LARGE";
+  "CURLOPT_MAX_SEND_SPEED_LARGE";
+  "CURLOPT_MIMEPOST";
+  "CURLOPT_NETRC";
+  "CURLOPT_NETRC_FILE";
+  "CURLOPT_NEW_DIRECTORY_PERMS";
+  "CURLOPT_NEW_FILE_PERMS";
+  "CURLOPT_NOBODY";
+  "CURLOPT_NOPROGRESS";
+  "CURLOPT_NOSIGNAL";
+  "CURLOPT_PASSWORD";
+  "CURLOPT_PIPEWAIT";
+  "CURLOPT_PORT";
+  "CURLOPT_POST";
+  "CURLOPT_POST301";
+  "CURLOPT_POSTFIELDS";
+  "CURLOPT_POSTFIELDSIZE";
+  "CURLOPT_POSTFIELDSIZE_LARGE";
+  "CURLOPT_POSTQUOTE";
+  "CURLOPT_POSTREDIR";
+  "CURLOPT_PREQUOTE";
+  "CURLOPT_PREREQFUNCTION";
+  "CURLOPT_PROGRESSDATA";
+  "CURLOPT_PROGRESSFUNCTION";
+  "CURLOPT_PROTOCOLS";
+  "CURLOPT_PROXY";
+  "CURLOPT_PROXYAUTH";
+  "CURLOPT_PROXYPORT";
+  "CURLOPT_PROXYTYPE";
+  "CURLOPT_PROXYUSERPWD";
+  "CURLOPT_PROXY_TRANSFER_MODE";
+  "CURLOPT_PUT";
+  "CURLOPT_QUOTE";
+  "CURLOPT_RANDOM_FILE";
+  "CURLOPT_RANGE";
+  "CURLOPT_READFUNCTION";
+  "CURLOPT_REDIR_PROTOCOLS";
+  "CURLOPT_REFERER";
+  "CURLOPT_RESOLVE";
+  "CURLOPT_RESUME_FROM";
+  "CURLOPT_RESUME_FROM_LARGE";
+  "CURLOPT_SEEKFUNCTION";
+  "CURLOPT_SHARE";
+  "CURLOPT_SSH_AUTH_TYPES";
+  "CURLOPT_SSH_HOST_PUBLIC_KEY_MD5";
+  "CURLOPT_SSH_PRIVATE_KEYFILE";
+  "CURLOPT_SSH_PUBLIC_KEYFILE";
+  "CURLOPT_SSLCERT";
+  "CURLOPT_SSLCERTPASSWD";
+  "CURLOPT_SSLCERTTYPE";
+  "CURLOPT_SSLENGINE";
+  "CURLOPT_SSLENGINE_DEFAULT";
+  "CURLOPT_SSLKEY";
+  "CURLOPT_SSLKEYPASSWD";
+  "CURLOPT_SSLKEYTYPE";
+  "CURLOPT_SSLVERSION";
+  "CURLOPT_SSL_CIPHER_LIST";
+  "CURLOPT_SSL_OPTIONS";
+  "CURLOPT_SSL_SESSIONID_CACHE";
+  "CURLOPT_SSL_VERIFYHOST";
+  "CURLOPT_SSL_VERIFYPEER";
+  "CURLOPT_TCP_FASTOPEN";
+  "CURLOPT_TCP_NODELAY";
+  "CURLOPT_TELNETOPTIONS";
+  "CURLOPT_TIMECONDITION";
+  "CURLOPT_TIMEOUT";
+  "CURLOPT_TIMEOUT_MS";
+  "CURLOPT_TIMEVALUE";
+  "CURLOPT_TRANSFERTEXT";
+  "CURLOPT_UNRESTRICTED_AUTH";
+  "CURLOPT_UPLOAD";
+  "CURLOPT_URL";
+  "CURLOPT_USERAGENT";
+  "CURLOPT_USERNAME";
+  "CURLOPT_USERPWD";
+  "CURLOPT_VERBOSE";
+  "CURLOPT_WRITEFUNCTION";
+  "CURLOPT_WRITEHEADER";
+  "CURLOPT_WRITEINFO";
+  "CURLOPT_XFERINFOFUNCTION";
+
+  "CURLSSLBACKEND_BEARSSL";
+  "CURLSSLBACKEND_GNUTLS";
+  "CURLSSLBACKEND_GSKIT";
+  "CURLSSLBACKEND_MBEDTLS";
+  "CURLSSLBACKEND_MESALINK";
+  "CURLSSLBACKEND_NONE";
+  "CURLSSLBACKEND_NSS";
+  "CURLSSLBACKEND_OPENSSL";
+  "CURLSSLBACKEND_SCHANNEL";
+  "CURLSSLBACKEND_SECURETRANSPORT";
+  "CURLSSLBACKEND_WOLFSSL";
+
+  "CURL_HTTP_VERSION_2";
+  "CURL_HTTP_VERSION_2TLS";
+  "CURL_HTTP_VERSION_2_0";
+  "CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE";
+  "CURL_HTTP_VERSION_3";
+
+  "CURL_SSLVERSION_TLSv1_0";
+  "CURL_SSLVERSION_TLSv1_1";
+  "CURL_SSLVERSION_TLSv1_2";
+  "CURL_SSLVERSION_TLSv1_3";
+
+  "CURL_VERSION_NTLM_WB";
+  "CURL_VERSION_TLSAUTH_SRP";
+
+  "curl_multi_poll";
+  "curl_global_sslset";
+]
+
+let extract_declarations c ~cflags ~libs =
+  let c_source decls =
+    let lines =
+      [
+        "#include <curl/curl.h>";
+        "int main()";
```

Comment:
```suggestion
        "int main(void)";
```

Compilers may warn against this obsolete form of function declaration (until C23 iirc).